components/ILIAS/Forum/classes/Tables/ForumStatisticsTable.php

```diff
@@ -0,0 +1,201 @@
+                                    $this
+                                )
+                                ->withRequest($this->request);
+    }
+
+    protected function getColumns(): array
+    {
+        $columns = [
+            'ranking' => $this->ui_factory->table()->column()->number(
+                $this->lng->txt('frm_statistics_ranking')
+            )->withIsSortable(true),
+            'login' => $this->ui_factory->table()->column()->text(
+                $this->lng->txt('login')
+            )->withIsSortable(true),
+            'lastname' => $this->ui_factory->table()->column()->text(
+                $this->lng->txt('lastname')
+            )->withIsSortable(true),
+            'firstname' => $this->ui_factory->table()->column()->text(
+                $this->lng->txt('firstname')
+            )->withIsSortable(true),
+        ];
+        if ($this->has_active_lp && $this->has_general_lp_access) {
+            $columns['progress'] = $this->ui_factory->table()->column()->status(
+                $this->lng->txt('learning_progress')
+            )->withIsSortable(false);
+        }
+        return $columns;
+    }
+
+    private function initRecords(): void
+    {
+        if ($this->records === null) {
+            $this->records = [];
+            $data = $this->forum->Forum->getUserStatistics($this->obj_properties->isPostActivationEnabled());
+            $result = [];
+            $counter = 0;
+            foreach ($data as $row) {
+                $this->records[$counter]['usr_id'] = $row['usr_id'];
+                $this->records[$counter]['ranking'] = $row['num_postings'];
+                $this->records[$counter]['login'] = $row['login'];
+                $this->records[$counter]['lastname'] = $row['lastname'];
+                $this->records[$counter]['firstname'] = $row['firstname'];
+                if ($this->has_active_lp && $this->has_general_lp_access) {
+                    $this->records[$counter]['progress'] = $this->getProgressStatus($row['usr_id']);
+                }
+                ++$counter;
+            }
+        }
+    }
+
+    private function sortedRecords(array $records, Order $order): array
+    {
+        [$order_field, $order_direction] = $order->join([], fn($ret, $key, $value) => [$key, $value]);
+
+        usort($records, static function ($left, $right) use ($order_field): int {
+            if ($order_field === 'ranking') {
+                return $left[$order_field] <=> $right[$order_field];
+            }
+            return ilStr::strCmp($left[$order_field], $right[$order_field]);
+        });
+
+        if ($order_direction === 'DESC') {
+            $records = array_reverse($records);
+        }
+        return $records;
+    }
+
+    private function getRecords(Range $range, Order $order): array
+    {
+        $this->initRecords();
+        $records = $this->sortedRecords($this->records, $order);
+        return $this->limitRecords($records, $range);
```

Comment:
Could we add an empty line here?

```suggestion

        return $this->limitRecords($records, $range);
```